core/src/main/scala/tofu/lift/Unlift.scala

```diff
@@ -118,11 +109,32 @@ object Unlift extends UnliftInstances1 {
 }
 
 private[lift] trait UnliftInstances1 {
-  implicit def unliftReaderT[F[_]: Applicative, R]: Unlift[F, ReaderT[F, R, *]] = {
-    type RT[a] = ReaderT[F, R, a]
-    new Unlift[F, RT] {
-      def lift[A](fa: F[A]): RT[A] = ReaderT.liftF(fa)
-      def unlift: RT[RT ~> F]      = ReaderT(r => funK[RT, F](_.run(r)).pure[F])
+  implicit def unliftWithRun[F[_], G[_], A](implicit wr: WithRun[F, G, A]): Unlift[G, F] = wr
+}
+
+private[lift] trait UnliftInstances2 {
+  implicit def unliftIOWithRun[F[_], G[_], A](implicit
```

Comment:
Very strange chain. It's just a composition of 2 Unlifts. You require WithRun, but use it as Unlift.

Reply:
we got _diverging implicit expansion_ if we request for 2 unlift`s, but with _WithRun_ we nicely derive UnliftIO for readers if base effect has UnliftIO or Effect. Small example https://scastie.scala-lang.org/FjgvC5IbSKitKLLAwMlK1g

Reply:
Is this a _real_ use case when you have `ReaderT[ReaderT[IO, R1, *], R2, *]`?
You'd better use `(R1, R2)` as an environment and optics.

Reply:
I consider this instance redundant. It is too specific (IO) and uses too powerful typeclass (WithRun). It may be replaced just with Unlift composition:
`(Unlift[F, G], Unlift[G, H], FlatMap[H]) => Unlift[F, H]`

If you have diverging implicits, this instance shouldn't be implicit as many other composition instances.

Reply:
1. `ReaderT[ReaderT[IO, R1, *], R2, *]` is not real use case, just an example of how we can derive `UnliftIO`. In my case i have `I[_]: Effect` and locally defined `type App[A] = ReaderT[I, Ctx, A]` (or `App[_]: WithRun[*[_], I, Ctx]` in _context bounds_) and in some places i want `UnliftIO` instance for `App`. I can make it work by hand with composition, but my intention was to derive it implicitly without _diverging implicits_. I know about `Unlift[F, G].effect` and that we can create `Effect` instance for readers, but you need to write it by hand and it adds a bit boilerplate. So in cases when are happy with `F[F~>IO]`, we can get it for free. 
2. Unlift composition already exist by non-implicit `andThen` method in `Unlift`.

Reply:
I think in your case it should work right now, without extra changes.